src/common/hooks/useInstance.js

```diff
@@ -1,30 +1,24 @@
-import {
-  useState,
-  useEffect,
-} from 'react';
+import useSearchInstanceByIdQuery from './useSearchInstanceByIdQuery';
+import useInstanceQuery from './useInstanceQuery';
 
-const useInstance = (id, mutator) => {
-  const [instance, setInstance] = useState();
-  const [isLoading, setIsLoading] = useState(true);
+const useInstance = (id) => {
+  const { instance: _instance } = useSearchInstanceByIdQuery(id);
 
-  useEffect(() => {
-    setIsLoading(true);
+  const instanceTenantId = _instance?.tenantId;
+  const isShared = _instance?.shared;
 
-    mutator.GET({
-      params: {
-        query: `id==${id}`,
-      },
-    })
-      .then((instances) => {
-        setInstance(instances[0]);
-      })
-      .finally(() => {
-        setIsLoading(false);
-      });
-  }, [id]);
+  const { isLoading, instance } = useInstanceQuery(
+    id,
+    { tenantId: instanceTenantId },
+    { enabled: Boolean(id && instanceTenantId) }
+  );
 
   return {
-    instance,
+    instance: {
```

Comment:
It used to be the same object on every render, and we could use it in dependencies e.g. in `useEffect`, now it is always a new object. Let's memo it to avoid unexpected fires of `useEffect` callback or excessive memoization, in case of using it in `useMemo` or `useCallback`.